http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/WhitelistAuthProvider.scala

```diff
@@ -25,62 +26,60 @@ class WhitelistAuthProvider(config: Config, saProvider: ServiceAccountProvider[I
   ): IO[Boolean] =
     checkWhitelist(userInfo)
 
-  override def hasNotebookClusterPermission(
-    internalId: RuntimeInternalId,
+  override def hasRuntimePermission(
+    samResource: RuntimeSamResource,
     userInfo: UserInfo,
-    action: NotebookClusterAction,
-    googleProject: GoogleProject,
-    runtimeName: RuntimeName
+    action: RuntimeAction,
+    googleProject: GoogleProject
   )(implicit ev: ApplicativeAsk[IO, TraceId]): IO[Boolean] =
     checkWhitelist(userInfo)
 
   override def hasPersistentDiskPermission(
-    internalId: DiskSamResourceId,
+    samResource: PersistentDiskSamResource,
     userInfo: UserInfo,
     action: PersistentDiskAction,
     googleProject: GoogleProject
   )(implicit ev: ApplicativeAsk[IO, TraceId]): IO[Boolean] =
     checkWhitelist(userInfo)
 
-  override def filterUserVisibleClusters(userInfo: UserInfo, clusters: List[(GoogleProject, RuntimeInternalId)])(
+  override def filterUserVisibleRuntimes(userInfo: UserInfo, clusters: List[(GoogleProject, RuntimeSamResource)])(
     implicit ev: ApplicativeAsk[IO, TraceId]
-  ): IO[List[(GoogleProject, RuntimeInternalId)]] =
+  ): IO[List[(GoogleProject, RuntimeSamResource)]] =
     clusters.traverseFilter { a =>
       checkWhitelist(userInfo).map {
         case true  => Some(a)
         case false => None
       }
     }
 
-  override def filterUserVisiblePersistentDisks(userInfo: UserInfo, disks: List[(GoogleProject, DiskSamResourceId)])(
+  override def filterUserVisiblePersistentDisks(userInfo: UserInfo,
+                                                disks: List[(GoogleProject, PersistentDiskSamResource)])(
     implicit ev: ApplicativeAsk[IO, TraceId]
-  ): IO[List[(GoogleProject, DiskSamResourceId)]] =
+  ): IO[List[(GoogleProject, PersistentDiskSamResource)]] =
     disks.traverseFilter { a =>
       checkWhitelist(userInfo).map {
         case true  => Some(a)
         case false => None
       }
     }
 
-  def notifyClusterCreated(internalId: RuntimeInternalId,
-                           creatorEmail: WorkbenchEmail,
-                           googleProject: GoogleProject,
-                           runtimeName: RuntimeName)(implicit ev: ApplicativeAsk[IO, TraceId]): IO[Unit] = IO.unit
+  def notifyRuntimeCreated(samResource: RuntimeSamResource, creatorEmail: WorkbenchEmail, googleProject: GoogleProject)(
+    implicit ev: ApplicativeAsk[IO, TraceId]
+  ): IO[Unit] = IO.unit
 
-  def notifyClusterDeleted(internalId: RuntimeInternalId,
+  def notifyRuntimeDeleted(samResource: RuntimeSamResource,
```

Comment:
can we have `notifyResourceDeleted` now?

Reply:
Spoke to @rtitle about this - I think the original idea was to have `LeoAuthProvider` have resource-specific methods, but the `SamDAO` treats resources the same. Since what we do in `LeoAuthProvider` is pretty different between resources (cache check for runtimes, projects treated differently, etc) I don't think there's a huge benefit to condensing it into one method. What do you think?

Reply:
Yeah I had proposed resource-specific methods in `LeoAuthProvider` (which take specific types for the resource in question)

Reply:
Different methods sounds reasonable if they differ a lot